test/functional/test_framework/mininode.py

```diff
@@ -171,7 +171,7 @@ def _on_data(self):
                 if len(self.recvbuf) < 4:
                     return
                 if self.recvbuf[:4] != self.magic_bytes:
-                    raise ValueError("got garbage %s" % repr(self.recvbuf))
+                    raise ValueError("magic bytes mismatch: {} != {}".format(repr(self.magic_bytes), repr(self.recvbuf)))
```

Comment:
```suggestion
                    raise ValueError("magic bytes mismatch: received {} does not start with the expected {}".format(self.recvbuf, self.magic_bytes))
```

Reply:
Was going to suggest something like this.